query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/converters/PinotFunctionConverterTest.java

```diff
@@ -173,10 +199,70 @@ void convertsConcatFunction() {
         "CONCATSKIPNULL(foo,bar)",
         new PinotFunctionConverter()
             .convert(
+                testExecutionContext,
                 buildFunction(QUERY_FUNCTION_CONCAT, column1.toBuilder(), column2.toBuilder()),
                 this.mockArgumentConverter));
   }
 
+  @Test
+  void convertAvgRateFunction() {
+    Expression column1 = createColumnExpression("foo").build();
+    Expression column2 = createStringLiteralValueExpression("PT2S");
+
+    when(this.mockArgumentConverter.apply(column1)).thenReturn("foo");
+    when(this.mockingExecutionContext.getTimeSeriesPeriod())
+        .thenReturn(Optional.of(Duration.ofSeconds(10)));
+
+    assertEquals(
+        "SUM(DIV(foo, 5.0))",
+        new PinotFunctionConverter()
+            .convert(
+                mockingExecutionContext,
+                buildFunction(QUERY_FUNCTION_AVG_RATE, column1.toBuilder(), column2.toBuilder()),
+                this.mockArgumentConverter));
+
+    when(this.mockingExecutionContext.getTimeSeriesPeriod()).thenReturn(Optional.empty());
+    when(this.mockingExecutionContext.getTimeRangeDuration()).thenReturn(Duration.ofSeconds(10));
+
+    assertEquals(
+        "SUM(DIV(foo, 5.0))",
+        new PinotFunctionConverter()
+            .convert(
+                mockingExecutionContext,
+                buildFunction(QUERY_FUNCTION_AVG_RATE, column1.toBuilder(), column2.toBuilder()),
+                this.mockArgumentConverter));
+
+    when(this.mockingExecutionContext.getTimeSeriesPeriod())
+        .thenReturn(Optional.of(Duration.ofSeconds(20)));
+    when(this.mockingExecutionContext.getTimeRangeDuration()).thenReturn(Duration.ofSeconds(10));
```

Comment:
nit: it would never make sense to have a duration < period unless I'm misunderstanding. Can make this one bigger (and also would be helpful to the reader to make it unambiguously different than the result of 20/2)